docs/content/querying/groupbyquery.md

```diff
@@ -214,3 +214,4 @@ When using the "v2" strategy, the following query context parameters apply:
 |`maxMergingDictionarySize`|Can be used to lower the value of `druid.query.groupBy.maxMergingDictionarySize` for this query.|
 |`maxOnDiskStorage`|Can be used to lower the value of `druid.query.groupBy.maxOnDiskStorage` for this query.|
 |`sortByDimsFirst`|Sort the results first by dimension values and then by timestamp.|
+|`pushDownLimit`|When a limit spec is present, enables an optimization that applies the limit at the merge buffer level, reducing the number of results that need to be processed at the broker level. Note that when the sorting order uses fields that are not in the grouping key, this can result in approximate results.|
```

Comment:
The approximate results seem to mean that the group-by results may not be valid if non-grouping keys are used for sort. This is because of the behavior of ```BufferGrouper```. It first aggregates input rows until its hash table fills. When the hash table fills, it copies the top-n entries found so far to another hash table, and continues aggregation. Here, the copied entires are able to not belong to the real group-by result, and this can make the result invalid.

In this case, we cannot guarantee the approximation ratio because it will depend on the input order. In the worst case, users can get totally unexpected results but it's difficult for them to figure out until running the same query without limit push down. 

So, IMHO, it would be better to disable when non-grouping keys are used for sort. What do you think?